node/grpc/server_v2.go

```diff
@@ -126,23 +114,36 @@ func (s *ServerV2) StoreChunks(ctx context.Context, in *pb.StoreChunksRequest) (
 
 	batch, err := s.validateStoreChunksRequest(in)
 	if err != nil {
-		return nil, err
+		return nil, api.NewErrorInvalidArg(fmt.Sprintf("failed to validate store chunk request: %v", err))
```

Comment:
Based on the PR description, I guess validateStoreChunksRequest is a cheaper operation than authenticating the request, hence the ordering change? Could you add a comment describing that.

Reply:
yes. we shouldn't spend resource on deep processing a request if it's invalid, i.e. reject invalid request with least amount of resource, which is helpful for security (e.g. spam requests)